src/main/java/org/icgc/argo/workflow_management/service/NextflowService.java

```diff
@@ -36,10 +37,19 @@
 @Slf4j
 @Service(value = "nextflow")
 public class NextflowService implements WorkflowExecutionService {
-
-  @Autowired private NextflowProperties config;
-
-  private Scheduler scheduler = Schedulers.newElastic("nextflow-service");
+  private final NextflowProperties config;
+  private final WorkflowStatusMonitor statusMonitor;
+  private final Scheduler scheduler;
+
+  @Autowired
+  public NextflowService(NextflowProperties config) {
+    this.config = config;
+    this.scheduler = Schedulers.newElastic("nextflow-service");
+    this.statusMonitor = new WorkflowStatusMonitor(config);
```

Comment:
As mentioned in the other comment, as this is your runnable, and the default scope for IoC in Spring is Singleton, this constructor is only ever invoked once, and thus you will only ever have on instance of the thread unless I have missed something obvious.